tests/main/snap-remove-terminate/task.yaml

```diff
@@ -25,7 +25,7 @@ execute: |
     retry -n 10 systemctl is-active test-kill.service
 
     echo "Lock is held"
-    not flock --timeout 0 "$lockfile" --command "true"
+    not flock --timeout 10 "$lockfile" --command "true"
```

Comment:
is this correct though? we're expecting the code running in unit to to take the lock, so if instead it is possible for the test to do so, something else isn't right, maybe the unit isn't running? AFAIU the --timeout only extends the time flock will wait to acquire the lock.

How about we refactor the test such that test-snapd-sh.sh executes eg: `sh -c 'touch $SNAP_COMMON/alive; sleep 10000', and then the test could wait for the file to show up and double check that /sys/fs/{unified/,}cgroup/system.slice/test-kill.service/cgroup.procs has at least one line (pid)

Reply:
This much better, Thanks! but I don't know if we need to check cgroup pids in the test though? if test-snapd-sh.sh created the file it should be running

Reply:
Updated, Thank you!